--- a/oxygen-fonts-5.3.0/CMakeLists.txt	2015-05-06 02:14:32.822156477 +0200
+++ b/oxygen-fonts-5.3.0/CMakeLists.txt	2015-05-06 02:22:03.142985121 +0200
@@ -31,36 +31,5 @@ install(FILES
         DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
         COMPONENT Devel)
 
-if(NOT DEFINED OXYGEN_FONT_INSTALL_DIR)
-  set(OXYGEN_FONT_INSTALL_DIR  ${CMAKE_INSTALL_PREFIX}/${SHARE_INSTALL_PREFIX}/fonts/truetype/oxygen)
-endif()
-
-find_program(FONTFORGE_EXECUTABLE NAMES fontforge)
-if(FONTFORGE_EXECUTABLE)
-    message(STATUS "FontForge found: ${FONTFORGE_EXECUTABLE}")
-else()
-    message(FATAL_ERROR "Could not find FontForge executable")
-endif()
-
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
-add_custom_target(regular ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Regular-400/src/Oxygen-Sans.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans.ttf
-                  )
-add_custom_target(bold ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/Bold-700/src/Oxygen-Sans-Bold.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans-Bold.ttf
-                  )
-add_custom_target(mono ALL fontforge ${CMAKE_CURRENT_SOURCE_DIR}/generate-ttf.pe
-                   ${CMAKE_CURRENT_SOURCE_DIR}/oxygen-fonts/mono-400/src/OxygenMono-Regular.sfd
-                   ${CMAKE_CURRENT_BINARY_DIR}/OxygenMono-Regular.ttf
-                  )
-
-set(fontFiles
-    ${CMAKE_CURRENT_BINARY_DIR}/OxygenMono-Regular.ttf
-    ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans-Bold.ttf
-    ${CMAKE_CURRENT_BINARY_DIR}/Oxygen-Sans.ttf
-)
-
-install(FILES ${fontFiles} DESTINATION ${OXYGEN_FONT_INSTALL_DIR})
