TuxJDK
===================================================================
--- a/src/share/classes/java/awt/font/FontRenderContext.java
+++ b/src/share/classes/java/awt/font/FontRenderContext.java
@@ -75,7 +75,7 @@ public class FontRenderContext {
      *
      */
     protected FontRenderContext() {
-        aaHintValue = VALUE_TEXT_ANTIALIAS_DEFAULT;
+        aaHintValue = VALUE_TEXT_ANTIALIAS_LCD_HRGB;
         fmHintValue = VALUE_FRACTIONALMETRICS_DEFAULT;
         defaulting = true;
     }
===================================================================
--- a/src/share/classes/sun/java2d/SunGraphics2D.java
+++ b/src/share/classes/sun/java2d/SunGraphics2D.java
@@ -278,7 +278,7 @@ public final class SunGraphics2D
 
         renderHint = SunHints.INTVAL_RENDER_DEFAULT;
         antialiasHint = SunHints.INTVAL_ANTIALIAS_OFF;
-        textAntialiasHint = SunHints.INTVAL_TEXT_ANTIALIAS_DEFAULT;
+        textAntialiasHint = SunHints.INTVAL_TEXT_ANTIALIAS_LCD_HRGB;
         fractionalMetricsHint = SunHints.INTVAL_FRACTIONALMETRICS_OFF;
         lcdTextContrast = lcdTextContrastDefaultValue;
         interpolationHint = -1;
@@ -1354,7 +1354,7 @@ public final class SunGraphics2D
         this.hints = null;
         renderHint = SunHints.INTVAL_RENDER_DEFAULT;
         antialiasHint = SunHints.INTVAL_ANTIALIAS_OFF;
-        textAntialiasHint = SunHints.INTVAL_TEXT_ANTIALIAS_DEFAULT;
+        textAntialiasHint = SunHints.INTVAL_TEXT_ANTIALIAS_LCD_HRGB;
         fractionalMetricsHint = SunHints.INTVAL_FRACTIONALMETRICS_OFF;
         lcdTextContrast = lcdTextContrastDefaultValue;
         interpolationHint = -1;
