--- a/configure	2013-03-29 08:11:54.000000000 +0100
+++ b/configure	2013-05-12 16:50:11.850823431 +0200
@@ -657,6 +657,8 @@ BASECONFIGDIR
 baseconfigdir
 TEMPLATEDIR
 templatedir
+TEMPLATEINFDIR
+templateinfdir
 FC_FONTDATE
 FC_CACHEDIR
 fc_cachedir
@@ -835,6 +837,7 @@ with_default_fonts
 with_add_fonts
 with_cache_dir
 with_templatedir
+with_templateinfdir
 with_baseconfigdir
 with_configdir
 with_xmldir
@@ -1518,8 +1521,13 @@ Optional Packages:
                           Find additional fonts in DIR1,DIR2,...
   --with-cache-dir=DIR    Use DIR to store cache files
                           [default=LOCALSTATEDIR/cache/fontconfig]
-  --with-templatedir=DIR  Use DIR to store the configuration template files
+  --with-templatedir=DIR  Use DIR to store the generic configuration 
+                          template files
                           [default=DATADIR/fontconfig/conf.avail]
+  --with-templateinfdir=DIR
+                          Use DIR to store the Infinality compliant 
+                          configuration template files
+                          [default=DATADIR/fontconfig/conf.avail.infinality]
   --with-baseconfigdir=DIR
                           Use DIR to store the base configuration files
                           [default=SYSCONFDIR/fonts]
@@ -14718,6 +14726,14 @@ else
 fi
 
 
+# Check whether --with-templateinfdir was given.
+if test "${with_templateinfdir+set}" = set; then :
+  withval=$with_templateinfdir; templateinfdir="$withval"
+else
+  templateinfdir=yes
+fi
+
+
 # Check whether --with-baseconfigdir was given.
 if test "${with_baseconfigdir+set}" = set; then :
   withval=$with_baseconfigdir; baseconfigdir="$withval"
@@ -14749,6 +14765,13 @@ no|yes)
 *)
 	;;
 esac
+case "$templateinfdir" in
+no|yes)
+	templateinfdir='${datadir}'/fontconfig/conf.avail.infinality
+	;;
+*)
+	;;
+esac
 case "$baseconfigdir" in
 no|yes)
 	baseconfigdir='${sysconfdir}'/fonts
@@ -14771,7 +14794,9 @@ no|yes)
 	;;
 esac
 
+
 TEMPLATEDIR=${templatedir}
+TEMPLATEINFDIR=${templateinfdir}
 BASECONFIGDIR=${baseconfigdir}
 CONFIGDIR=${configdir}
 XMLDIR=${xmldir}
@@ -14781,11 +14806,6 @@ XMLDIR=${xmldir}
 
 
 
-
-
-
-
-
 #
 # Thread-safety primitives
 #
@@ -15750,7 +15770,7 @@ _ACEOF
 
 
 
-ac_config_files="$ac_config_files Makefile fontconfig/Makefile fc-lang/Makefile fc-glyphname/Makefile fc-case/Makefile src/Makefile conf.d/Makefile fc-cache/Makefile fc-cat/Makefile fc-list/Makefile fc-match/Makefile fc-pattern/Makefile fc-query/Makefile fc-scan/Makefile fc-validate/Makefile doc/Makefile doc/version.sgml test/Makefile fontconfig.spec fontconfig.pc fontconfig-zip"
+ac_config_files="$ac_config_files Makefile fontconfig/Makefile fc-lang/Makefile fc-glyphname/Makefile fc-case/Makefile src/Makefile conf.d/Makefile conf.d.infinality/Makefile fc-cache/Makefile fc-cat/Makefile fc-list/Makefile fc-match/Makefile fc-pattern/Makefile fc-query/Makefile fc-scan/Makefile fc-validate/Makefile doc/Makefile doc/version.sgml test/Makefile fontconfig.spec fontconfig.pc fontconfig-zip"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -16814,6 +16834,7 @@ do
     "fc-case/Makefile") CONFIG_FILES="$CONFIG_FILES fc-case/Makefile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "conf.d/Makefile") CONFIG_FILES="$CONFIG_FILES conf.d/Makefile" ;;
+    "conf.d.infinality/Makefile") CONFIG_FILES="$CONFIG_FILES conf.d.infinality/Makefile" ;;
     "fc-cache/Makefile") CONFIG_FILES="$CONFIG_FILES fc-cache/Makefile" ;;
     "fc-cat/Makefile") CONFIG_FILES="$CONFIG_FILES fc-cat/Makefile" ;;
     "fc-list/Makefile") CONFIG_FILES="$CONFIG_FILES fc-list/Makefile" ;;
