--- a/configure	2014-03-24 07:04:16.000000000 +0100
+++ b/configure	2015-03-18 21:26:18.805031769 +0100
@@ -655,6 +655,7 @@ XMLDIR
 CONFIGDIR
 BASECONFIGDIR
 TEMPLATEDIR
+TEMPLATEINFDIR
 FC_FONTDATE
 FC_CACHEDIR
 fc_cachedir
@@ -833,6 +834,7 @@ with_default_fonts
 with_add_fonts
 with_cache_dir
 with_templatedir
+with_templateinfdir
 with_baseconfigdir
 with_configdir
 with_xmldir
@@ -1518,6 +1520,10 @@ Optional Packages:
                           [default=LOCALSTATEDIR/cache/fontconfig]
   --with-templatedir=DIR  Use DIR to store the configuration template files
                           [default=DATADIR/fontconfig/conf.avail]
+  --with-templateinfdir=DIR
+                          Use DIR to store the Infinality compliant
+                          configuration template files
+                          [default=DATADIR/fontconfig/conf.avail.infinality]
   --with-baseconfigdir=DIR
                           Use DIR to store the base configuration files
                           [default=SYSCONFDIR/fonts]
@@ -14856,6 +14862,12 @@ else
   templatedir=yes
 fi
 
+# Check whether --with-templateinfdir was given.
+if test "${with_templateinfdir+set}" = set; then :
+  withval=$with_templateinfdir; templateinfdir="$withval"
+else
+  templateinfdir=yes
+fi
 
 # Check whether --with-baseconfigdir was given.
 if test "${with_baseconfigdir+set}" = set; then :
@@ -14888,6 +14900,13 @@ no|yes)
 *)
 	;;
 esac
+case "$templateinfdir" in
+no|yes)
+	templateinfdir='${datadir}'/fontconfig/conf.avail.infinality
+	;;
+*)
+	;;
+esac
 case "$baseconfigdir" in
 no|yes)
 	baseconfigdir='${sysconfdir}'/fonts
@@ -14911,6 +14930,7 @@ no|yes)
 esac
 
 TEMPLATEDIR=${templatedir}
+TEMPLATEINFDIR=${templateinfdir}
 BASECONFIGDIR=${baseconfigdir}
 CONFIGDIR=${configdir}
 XMLDIR=${xmldir}
@@ -15939,7 +15959,7 @@ _ACEOF
 
 
 
-ac_config_files="$ac_config_files Makefile fontconfig/Makefile fc-lang/Makefile fc-glyphname/Makefile fc-case/Makefile src/Makefile conf.d/Makefile fc-cache/Makefile fc-cat/Makefile fc-list/Makefile fc-match/Makefile fc-pattern/Makefile fc-query/Makefile fc-scan/Makefile fc-validate/Makefile doc/Makefile doc/version.sgml test/Makefile fontconfig.spec fontconfig.pc fontconfig-zip"
+ac_config_files="$ac_config_files Makefile fontconfig/Makefile fc-lang/Makefile fc-glyphname/Makefile fc-case/Makefile src/Makefile conf.d/Makefile conf.d.infinality/Makefile fc-cache/Makefile fc-cat/Makefile fc-list/Makefile fc-match/Makefile fc-pattern/Makefile fc-query/Makefile fc-scan/Makefile fc-validate/Makefile doc/Makefile doc/version.sgml test/Makefile fontconfig.spec fontconfig.pc fontconfig-zip"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
