--- a/configure.ac	2013-10-11 06:27:42.000000000 +0200
+++ b/configure.ac	2013-10-11 19:02:55.440016339 +0200
@@ -553,9 +553,14 @@ AC_SUBST(FC_FONTDATE)
 
 AC_ARG_WITH(templatedir,
 	[AC_HELP_STRING([--with-templatedir=DIR],
-			[Use DIR to store the configuration template files [default=DATADIR/fontconfig/conf.avail]])],
+			[Use DIR to store the generic configuration template files [default=DATADIR/fontconfig/conf.avail]])],
 	[templatedir="$withval"],
 	[templatedir=yes])
+AC_ARG_WITH(templateinfdir,
+	[AC_HELP_STRING([--with-templateinfdir=DIR],
+			[Use DIR to store the Infinality compliant configuration template files [default=DATADIR/fontconfig/conf.avail.infinality]])],
+	[templateinfdir="$withval"],
+	[templateinfdir=yes])
 AC_ARG_WITH(baseconfigdir,
 	[AC_HELP_STRING([--with-baseconfigdir=DIR],
 			[Use DIR to store the base configuration files [default=SYSCONFDIR/fonts]])],
@@ -579,6 +584,13 @@ no|yes)
 *)
 	;;
 esac
+case "$templateinfdir" in
+no|yes)
+	templateinfdir='${datadir}'/fontconfig/conf.avail.infinality
+	;;
+*)
+	;;
+esac
 case "$baseconfigdir" in
 no|yes)
 	baseconfigdir='${sysconfdir}'/fonts
@@ -602,10 +614,12 @@ no|yes)
 esac
 
 TEMPLATEDIR=${templatedir}
+TEMPLATEINFDIR=${templateinfdir}
 BASECONFIGDIR=${baseconfigdir}
 CONFIGDIR=${configdir}
 XMLDIR=${xmldir}
 AC_SUBST(TEMPLATEDIR)
+AC_SUBST(TEMPLATEINFDIR)
 AC_SUBST(BASECONFIGDIR)
 AC_SUBST(CONFIGDIR)
 AC_SUBST(XMLDIR)
@@ -733,6 +747,7 @@ fc-glyphname/Makefile
 fc-case/Makefile
 src/Makefile
 conf.d/Makefile
+conf.d.infinality/Makefile
 fc-cache/Makefile
 fc-cat/Makefile
 fc-list/Makefile
