--- a/configure.ac	2016-04-06 14:05:36.000000000 +0200
+++ b/configure.ac	2016-04-10 05:01:11.577597555 +0200
@@ -524,9 +524,14 @@ AC_SUBST(FC_FONTDATE)
 
 AC_ARG_WITH(templatedir,
 	[AC_HELP_STRING([--with-templatedir=DIR],
-			[Use DIR to store the configuration template files [default=DATADIR/fontconfig/conf.avail]])],
+			[Use DIR to store the generic configuration template files [default=DATADIR/fontconfig/conf.avail]])],
 	[templatedir="$withval"],
 	[templatedir=yes])
+AC_ARG_WITH(templateinfdir,
+	[AC_HELP_STRING([--with-templateinfdir=DIR],
+			[Use DIR to store the Infinality compliant configuration template files [default=DATADIR/fontconfig/conf.avail.infinality]])],
+	[templateinfdir="$withval"],
+	[templateinfdir=yes])
 AC_ARG_WITH(baseconfigdir,
 	[AC_HELP_STRING([--with-baseconfigdir=DIR],
 			[Use DIR to store the base configuration files [default=SYSCONFDIR/fonts]])],
@@ -550,6 +555,13 @@ no|yes)
 *)
 	;;
 esac
+case "$templateinfdir" in
+no|yes)
+	templateinfdir='${datadir}'/fontconfig/conf.avail.infinality
+	;;
+*)
+	;;
+esac
 case "$baseconfigdir" in
 no|yes)
 	baseconfigdir='${sysconfdir}'/fonts
@@ -573,10 +585,12 @@ no|yes)
 esac
 
 TEMPLATEDIR=${templatedir}
+TEMPLATEINFDIR=${templateinfdir}
 BASECONFIGDIR=${baseconfigdir}
 CONFIGDIR=${configdir}
 XMLDIR=${xmldir}
 AC_SUBST(TEMPLATEDIR)
+AC_SUBST(TEMPLATEINFDIR)
 AC_SUBST(BASECONFIGDIR)
 AC_SUBST(CONFIGDIR)
 AC_SUBST(XMLDIR)
@@ -706,6 +720,7 @@ fc-blanks/Makefile
 fc-case/Makefile
 src/Makefile
 conf.d/Makefile
+conf.d.infinality/Makefile
 fc-cache/Makefile
 fc-cat/Makefile
 fc-list/Makefile
