--- a/configure.ac	2013-03-29 08:10:15.000000000 +0100
+++ b/configure.ac	2013-05-12 12:32:24.493474542 +0200
@@ -514,9 +514,14 @@ AC_SUBST(FC_FONTDATE)
 
 AC_ARG_WITH(templatedir,
 	[AC_HELP_STRING([--with-templatedir=DIR],
-			[Use DIR to store the configuration template files [default=DATADIR/fontconfig/conf.avail]])],
+			[Use DIR to store the generic configuration template files [default=DATADIR/fontconfig/conf.avail]])],
 	[templatedir="$withval"],
 	[templatedir=yes])
+AC_ARG_WITH(templateinfdir,
+	[AC_HELP_STRING([--with-templateinfdir=DIR],
+			[Use DIR to store the Infinality compliant configuration template files [default=DATADIR/fontconfig/conf.avail.infinality]])],
+	[templateinfdir="$withval"],
+	[templateinfdir=yes])
 AC_ARG_WITH(baseconfigdir,
 	[AC_HELP_STRING([--with-baseconfigdir=DIR],
 			[Use DIR to store the base configuration files [default=SYSCONFDIR/fonts]])],
@@ -540,6 +545,13 @@ no|yes)
 *)
 	;;
 esac
+case "$templateinfdir" in
+no|yes)
+	templateinfdir='${datadir}'/fontconfig/conf.avail.infinality
+	;;
+*)
+	;;
+esac
 case "$baseconfigdir" in
 no|yes)
 	baseconfigdir='${sysconfdir}'/fonts
@@ -563,11 +575,14 @@ no|yes)
 esac
 
 TEMPLATEDIR=${templatedir}
+TEMPLATEINFDIR=${templateinfdir}
 BASECONFIGDIR=${baseconfigdir}
 CONFIGDIR=${configdir}
 XMLDIR=${xmldir}
 AC_SUBST(templatedir)
 AC_SUBST(TEMPLATEDIR)
+AC_SUBST(templateinfdir)
+AC_SUBST(TEMPLATEINFDIR)
 AC_SUBST(baseconfigdir)
 AC_SUBST(BASECONFIGDIR)
 AC_SUBST(configdir)
@@ -692,6 +707,7 @@ fc-glyphname/Makefile
 fc-case/Makefile
 src/Makefile
 conf.d/Makefile
+conf.d.infinality/Makefile
 fc-cache/Makefile
 fc-cat/Makefile
 fc-list/Makefile
