--- a/Makefile.am	2014-01-20 09:14:19.000000000 +0100
+++ b/Makefile.am	2015-03-18 22:14:14.902609277 +0100
@@ -23,7 +23,7 @@
 
 SUBDIRS=fontconfig fc-blanks fc-case fc-lang fc-glyphname src \
 	fc-cache fc-cat fc-list fc-match fc-pattern fc-query fc-scan \
-	fc-validate conf.d test
+	fc-validate conf.d conf.d.infinality test
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
--- a/Makefile.in	2014-03-24 07:04:15.000000000 +0100
+++ b/Makefile.in	2015-03-18 22:19:27.809614806 +0100
@@ -138,6 +138,8 @@ AM_V_at = $(am__v_at_@AM_V@)
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
+depcomp =
+am__depfiles_maybe =
 SOURCES =
 DIST_SOURCES =
 RECURSIVE_TARGETS = all-recursive check-recursive cscopelist-recursive \
@@ -213,7 +215,7 @@ CTAGS = ctags
 CSCOPE = cscope
 DIST_SUBDIRS = fontconfig fc-case fc-lang fc-glyphname src fc-cache \
 	fc-cat fc-list fc-match fc-pattern fc-query fc-scan \
-	fc-validate conf.d test doc
+	fc-validate conf.d conf.d.infinality test doc
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -354,6 +356,7 @@ SET_MAKE = @SET_MAKE@
 SHELL = @SHELL@
 STRIP = @STRIP@
 TEMPLATEDIR = @TEMPLATEDIR@
+TEMPLATEINFDIR = @TEMPLATEINFDIR@
 VERSION = @VERSION@
 WARN_CFLAGS = @WARN_CFLAGS@
 XMLDIR = @XMLDIR@
@@ -415,7 +418,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = fontconfig fc-case fc-lang fc-glyphname src fc-cache fc-cat \
 	fc-list fc-match fc-pattern fc-query fc-scan fc-validate \
-	conf.d test $(am__append_1)
+	conf.d conf.d.infinality test $(am__append_1)
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = fontconfig.pc.in fonts.conf.in fonts.dtd \
 	fontconfig.spec.in fontconfig.spec fontconfig-zip.in \
