--- a/fonts.conf.in	2013-11-03 00:30:32.580157876 +0100
+++ b/fonts.conf.in	2013-11-03 00:38:55.580163350 +0100
@@ -21,7 +21,18 @@
 	Keith Packard
 -->
 
-<!-- Font directory list -->
+  <!-- Set DPI. dpi should be set in ~/.Xresources to 96 -->
+  <!-- Setting to 72 here makes the px to pt conversions work better 
+       (Chrome/Chromium). Some may need to set this to 96, though.
+       Uncomment the rule below in case of Chrome/Chromium px to pt 
+       conversion issues. -->
+  <!--match target="pattern">
+    <edit name="dpi" mode="assign">
+      <double>72</double>
+    </edit>
+  </match-->
+
+ <!-- Font directory list -->
 
 	<dir>@FC_DEFAULT_FONTS@</dir>
 	@FC_FONTPATH@
