--- a/conf.d/30-urw-aliases.conf	2009-08-06 16:23:57.000000000 +0100
+++ b/conf.d/30-urw-aliases.conf	2009-08-06 16:25:07.000000000 +0100
@@ -29,6 +29,10 @@
 	  <family>Zapf Dingbats</family>
 	  <accept><family>Dingbats</family></accept>
 	</alias>
+	<alias binding="same">
+	  <family>ZapfDingbats</family>
+	  <accept><family>Dingbats</family></accept>
+	</alias>
 	<match target="pattern">
 	  <test name="family">
 	    <string>Symbol</string>
